cmd/gitserver/server/server.go

```diff
@@ -1385,7 +1385,14 @@ func (s *Server) handleBatchLog(w http.ResponseWriter, r *http.Request) {
 		}
 
 		var buf bytes.Buffer
-		cmd := exec.CommandContext(ctx, "git", "log", "-n", "1", "--name-only", format, string(repoCommit.CommitID))
+
+		commitId := string(repoCommit.CommitID)
+		// make sure CommitID is not an arg
+		if commitId[0] == '-' {
+			return "", true, err
```

Comment:
`err` would be nil in this case, so this would be interpreted at callsite as "ok it worked the hash is `''`", I think you want to return a proper error 
```suggestion
		// make sure CommitID is not an arg
		if commitId[0] == '-' {
			return "", true, errors.New("commit ID starting with - is not allowed")
```